metric_learn/constraints.py

```diff
@@ -80,7 +84,11 @@ def chunks(self, num_chunks=100, chunk_size=2):
     return chunks
 
   @staticmethod
-  def random_subset(all_labels, num_preserved=np.inf):
+  def random_subset(all_labels, num_preserved=np.inf, random_state=None):
+    """
+    the random state object to be passed must be a numpy random seed
+    """
+    state = random_state
     n = len(all_labels)
     num_ignored = max(0, n - num_preserved)
     idx = np.random.randint(n, size=num_ignored)
```

Comment:
Okay, cool. I'll add the `np.random` as the default in all the other classes as well.